### PR TITLE
feat: added enum completion

### PR DESCRIPTION
**Pull request by rin-yato**
Please review carefully, Im not sure where to put helper functions and types. I wanted to break the function into smaller functions, so that it would be easier to read.

### CODE REVIEW

**Comment by rin-yato** [comment]
Oh no, should I have included the package.json in my pull request?

File: src/renderer/components/CodeEditor/SqlCodeEditor.tsx [hunk 1]
```diff
@@ -6,20 +6,94 @@ import {
...
+      // dont run if there is no enumSchema
+      if (enumSchema.length === 0) return null;
+
+      const isEqualOperator = tree.prevSibling;
+
+      // check if it's an operator and id is 15
+      if (
+        !(
+          isEqualOperator?.type.name === 'Operator' &&
+          isEqualOperator?.type.id === 15
```
Comment (invisal) [inline src/renderer/components/CodeEditor/SqlCodeEditor.tsx:49]: Is the ID of the `=` always 15? I think it is better to do.

```
context.doc.sliceString(isEqualOperator.from, isEqualOperator.to) === '='
```

And also remember that '=' and '!=' also trigger the ENUM auto completion

File: src/renderer/components/CodeEditor/SqlCodeEditor.tsx [hunk 2]
```diff
@@ -6,20 +6,94 @@ import {
...
+
+      // split the text to table and column
+      const [table, column] = tableAndColumnText.split('.');
+      // only check for table since column is possibly a keyword
+      if (!table) return null;
+
+      const enumValues = enumSchema.find((tempEnum) => {
+        if (column) {
+          // normally column will be enum
+          return tempEnum.column === column.replaceAll('`', '');
```
Comment (invisal) [inline src/renderer/components/CodeEditor/SqlCodeEditor.tsx:73]: It would be expensive to repeating replaceAll which can just do it one time at the top.

**Review by invisal — CHANGES_REQUESTED** [review]

File: src/renderer/components/CodeEditor/SqlCodeEditor.tsx (hunk 2 above)
Comment (rin-yato) [reply src/renderer/components/CodeEditor/SqlCodeEditor.tsx:73]: How did I not think of this? 😱

File: src/renderer/components/CodeEditor/SqlCodeEditor.tsx (hunk 1 above)
Comment (rin-yato) [reply src/renderer/components/CodeEditor/SqlCodeEditor.tsx:49]: the != is also considered as two separate node 💆‍♂️

File: src/renderer/components/CodeEditor/SqlCodeEditor.tsx (hunk 2 above)
Comment (rin-yato) [reply src/renderer/components/CodeEditor/SqlCodeEditor.tsx]: I have moved replaceAll higher up

File: src/renderer/components/CodeEditor/SqlCodeEditor.tsx (hunk 1 above)
Comment (rin-yato) [reply src/renderer/components/CodeEditor/SqlCodeEditor.tsx]: I have changed the logic to explicitly check for '=' and '!='

**Comment by rin-yato** [comment]
I have changed the code as per your suggestion.

**Pull request merged**